tests/basic/labels.rs

```diff
@@ -0,0 +1,337 @@
+use shuttle::{
+    check_dfs, check_random,
+    current::{get_label_for_task, me, set_label_for_task, set_name_for_task, LabelFn, TaskName},
+    future, thread,
+};
+use std::collections::HashSet;
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+use test_log::test;
+use tracing::field::{Field, Visit};
+use tracing::span::{Attributes, Record};
+use tracing::{Event, Id, Metadata, Subscriber};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct Ident(usize);
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct Parent(usize);
+
+// This test does the following: spawns a tree of tasks with integer ids
+// Each task has labels `Parent(i)` and `Ident(j)` where i is the parent's id, and j is the task's id.
+// Here is the tree of tasks spawned:
+//                    (1)
+//                /        \
+//             (2)          (3)
+//            /   \        /   \
+//         (4)    (5)    (6)   (7)
+//
+// Each leaf task increments a global AtomicUsize (initially 0) by 1
+// Each task returns its (Parent, Ident) as a usize pair when it completes.
+// The root waits for all tasks to complete, and collects the returned usize pairs. It then checks
+// - that the final AtomicUsize value is 4
+// - that each (parent, child) pair in the tree above is reported exactly once
+//
+// This test checks the following properties
+// - tasks inherit the Labels of their parent
+// - changing child Labels doesn't affect parent Labels
+// - child Labels are independent of each other
+async fn spawn_tasks(counter: Arc<AtomicUsize>) -> HashSet<(usize, usize)> {
+    set_label_for_task(me(), Ident(1));
```

Comment:
Not super important, but I find the `set_label_for_task(me()` slightly annoying (the annoying part is needing two imports/calls to do the most common operation for `Labels`). Not super fan of `set_label_for_current_task` either (though I think I prefer that over the `set me`. Maybe `set_label(label: Label)` ?

Reply:
Yeah, I didn't want to duplicate the function set with `{get,set}_label_for_task` and `{get,set}_label_for_current_task`.  One alternative I considered was to allow the argument to be `Option<TaskId>`, and making `None` always mean the current task.  I wasn't sure how readable that would be, and it's just as many characters as `me()`.